packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -104,9 +107,12 @@ function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
 		initial,
 	};
 
+	const classes = cx( css( { overflowX: 'auto' } ), className );
```

Comment:
Let's wrap the classname computation in a `useMemo` and move it earlier in the code, since the `className` should also be applied when `useReducedMotion` returns `false`:

<details>
<summary>See example</summary>

```diff
diff --git a/packages/components/src/navigator/navigator-screen/component.tsx b/packages/components/src/navigator/navigator-screen/component.tsx
index 26df748102..47a329c379 100644
--- a/packages/components/src/navigator/navigator-screen/component.tsx
+++ b/packages/components/src/navigator/navigator-screen/component.tsx
@@ -10,7 +10,7 @@ import { css } from '@emotion/react';
 /**
  * WordPress dependencies
  */
-import { useContext, useEffect, useState } from '@wordpress/element';
+import { useContext, useEffect, useState, useMemo } from '@wordpress/element';
 import { useReducedMotion, useFocusOnMount } from '@wordpress/compose';
 import { isRTL } from '@wordpress/i18n';
 
@@ -49,7 +49,12 @@ function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
 	const [ currentPath ] = useContext( NavigatorContext );
 	const isMatch = currentPath.path === path;
 	const ref = useFocusOnMount();
+
 	const cx = useCx();
+	const classes = useMemo(
+		() => cx( css( { overflowX: 'auto' } ), className ),
+		[ className ]
+	);
 
 	// This flag is used to only apply the focus on mount when the actual path changes.
 	// It avoids the focus to happen on the first render.
@@ -64,7 +69,7 @@ function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
 
 	if ( prefersReducedMotion ) {
 		return (
-			<View ref={ forwardedRef } { ...otherProps }>
+			<View ref={ forwardedRef } className={ classes } { ...otherProps }>
 				{ children }
 			</View>
 		);
@@ -107,8 +112,6 @@ function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
 		initial,
 	};
 
-	const classes = cx( css( { overflowX: 'auto' } ), className );
-
 	return (
 		<motion.div
 			ref={ hasPathChanged ? ref : undefined }
```

</details>

Reply:
Great catch, thank you!